code-challenges.test.js

```diff
@@ -22,25 +35,101 @@ const hitchhikersCharacters = [
+  it("takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.", () => {
+    expect(divRemainder(hodgepodge1)).toEqual([ 2, 0, -1, 0 ])
+    expect(divRemainder(hodgepodge2)).toEqual([ 2, 1, -1 ])
+  })
+})
+// ● divRemainder › takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+// ReferenceError: divRemainder is not defined
 
 const hodgepodge1 = [23, "Heyyyy!", 45, -10, 0, "Yo", false]
 // Expected output: [ 2, 0, -1, 0 ]
 const hodgepodge2 = [5, "Hola", 43, -34, "greetings", true]
 // Expected output: [ 2, 1, -1 ]
 
 // b) Create the function that makes the test pass.
+// Pseudocode
+// Input: An array with mixed data types
+// Output: An array of only the REMAINDERS of the numbers when divided by 3
+// Process:
+// Create a function called divRemainder
+// We know the outcome is going to be a subset of the original array of different length so we can use the filter method to iterate through the original array.
+// We only want nummbers so we can set a condition that we are keeping only the values that have the typeof 'number' or the value of 0 itself. 
+// Once we have the original numbers we can map over the new array and return each value modulus 3 to get the remainder of each value and update the array. 
+// Returns expected output.
+
+const divRemainder = (arr) => {
+    return arr.filter((value) => typeof value === 'number' || value === 0).map((value) => {return value % 3})
+}
+
+// PASS  ./code-challenges.test.js
+// divRemainder
+//   ✓ takes in a mixed data array and returns an array of only the REMAINDERS of the numbers when divided by 3.
+
+
 
 // --------------------3) Create a function that takes in an array of numbers and returns the sum of all the numbers cubed.
 
 // a) Create a test with an expect statement using the variables provided.
+describe("cubeSum", () => {
+  it("takes in an array of numbers and returns the sum of all the numbers cubed.", () => {
+    expect(cubeSum(cubeAndSum1)).toEqual(99)
+    expect(cubeSum(cubeAndSum2)).toEqual(1125)
+  })
+})
 
+// ● cubeSum › takes in an array of numbers and returns the sum of all the numbers cubed.
+
+// ReferenceError: cubeSum is not defined
 const cubeAndSum1 = [2, 3, 4]
 // Expected output: 99
 const cubeAndSum2 = [0, 5, 10]
 // Expected output: 1125
 
 // b) Create the function that makes the test pass.
+// Pseudocode
+// Input: An array of numbers
+// Output: The sum of all the numbers cubed as an integer
+// Process
+// Create a function called cubeSum
+// We can first iterate through the array using .map to replace the original value with the number cubed. **3
+// Lastly we need to return a single number as the sum of each value in the array. The reduce method returns a single value as the accumulated values of each element after the iterations are complete.. 
+
+
+const cubeSum = (arr) => {
+    return arr.map((value) => value ** 3).reduce((a,b) => a + b)
```

Comment:
Nice! You can also do this all within the reduce, removing the need for the map.
`arr.reduce((a,b) => a + b ** 3, 0)`